esindex/optimize.go

```diff
@@ -0,0 +1,27 @@
+package esindex
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+)
+
+// Optimize the target index (or indices) to have the maximum number of
+// segments. Segments < 1 will default to 1.
+//
+// Optimize blocks until the operation completes.
+func Optimize(target string, segn int) error {
+	resp, err := http.Post(fmt.Sprintf("%s/_optimize?max_num_segments=%d", target, segn), "text/plain", nil)
```

Comment:
I had no idea this existed..  It make this pattern very viable...  We may even be able to use this write pattern for our internal bulk write code.